db/python/tables/family.py

```diff
@@ -167,14 +179,60 @@ async def get_family_external_ids_by_participant_ids(
     async def update_family(
         self,
         id_: int,
-        external_id: str | None = None,
+        external_ids: dict[str, str | None] | None = None,
         description: str | None = None,
         coded_phenotype: str | None = None,
     ) -> bool:
         """Update values for a family"""
-        values: Dict[str, Any] = {'audit_log_id': await self.audit_log_id()}
-        if external_id:
-            values['external_id'] = external_id
+        audit_log_id = await self.audit_log_id()
+        values: Dict[str, Any] = {'audit_log_id': audit_log_id}
+
+        if external_ids:
+            to_delete = [k.lower() for k, v in external_ids.items() if v is None]
+            to_update = {k.lower(): v for k, v in external_ids.items() if v is not None}
+
+            if to_delete:
+                # Set audit_log_id to this transaction before deleting the rows
+                _audit_update_query = """
+UPDATE family_external_id
+SET audit_log_id = :audit_log_id
+WHERE family_id = :id AND name IN :names
+                """
+                await self.connection.execute(
+                    _audit_update_query,
+                    {'id': id_, 'names': to_delete, 'audit_log_id': audit_log_id},
+                )
+
+                _delete_query = """
+DELETE FROM family_exernal_id
```

Comment:
Turns out I didn't have test coverage on this function 🤣 

```suggestion
DELETE FROM family_external_id
```